lib/ioc-rc

```diff
@@ -543,10 +543,12 @@ __rc_jails () {
         local name="$(zfs get -H -o value org.freebsd.iocage:host_hostuuid \
                     $jail)"
         local boot="$(zfs get -H -o value org.freebsd.iocage:boot $jail)"
+        local _jail_hostid="$(zfs get -H -o value org.freebsd.iocage:hostid \
+                        $jail)"
         local priority="$(zfs get -H -o value org.freebsd.iocage:priority \
                         $jail)"
 
-        if [ "$boot" == "on" ] ; then
+        if [ "$boot" == "on" ] && [ "$_jail_hostid" == "$hostid" ] ; then
```

Comment:
No there is no real reason why I did it like this.
I am not a programmer and I only script if I need to. The reason I used
this is because I use that in my other scripts as well.



Op 14/09/15 om 23:57 schreef eli:
>
> In lib/ioc-rc
> <https://github.com/iocage/iocage/pull/222#discussion_r39452708>:
>
> >          local priority="$(zfs get -H -o value org.freebsd.iocage:priority \
> >                          $jail)"
> >  
> > -        if [ "$boot" == "on" ] ; then
> > +        if [ "$boot" == "on" ] && [ "$_jail_hostid" == "$hostid" ] ; then
>
> Is there a reason you are going with the |test| convention vs using
> |&&| and ||||?
> From what I understand, I believe |-a| and |-o| were "deprecated" from
> the posix standpoint (at least at one point).
>
> from: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html
>
>     The XSI extensions specifying the -a and -o binary primaries and
>     the '(' and ')' operators have been
>     marked obsolescent. (Many expressions using them are ambiguously
>     defined by the grammar
>     depending on the specific expressions being evaluated.) Scripts
>     using these expressions should be
>     converted to the forms given below. Even though many
>     implementations will continue to support these
>     obsolescent forms, scripts should be extremely careful when
>     dealing with user-supplied input that could
>     be confused with these and other primaries and operators. Unless
>     the application developer knows all
>     the cases that produce input to the script, invocations like:
>
>     test "$1" -a "$2"
>
>     should be written as:
>
>     |test "$1" && test "$2" |
>
> Not saying that your choice is wrong, or disagreeing with it. Just
> wondering if you could discuss why you chose to go that direction.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/iocage/iocage/pull/222/files#r39452708>.
>